File: client/src/entities/alert.ts
export type Alert = {
  alertId: number;
  locationId: number;
  deviceId: number;
  createdAt: number;
  severity: string;
  message: string;
  isResolved: boolean;
  resolvedAt: number;
};

export type AlertsResponse = {
  alerts: Alert[];
};

export type AlertWrapResponse = {
  alert: Alert;
};

export type AlertsQuery = {
  locationId?: number;
  from?: number;
  to?: number;
};

File: back/internal/domain/alert.go
package domain

type Alert struct {
	AlertID    int64
	LocationID int64
	DeviceID   int64
	CreatedAt  int64
	Severity   string
	Message    string
	IsResolved bool
	ResolvedAt int64
}

type AlertQuery struct {
	UserID           int64
	LocationID       int64
	FromCreatedAt    int64
	HasFromCreatedAt bool
	ToCreatedAt      int64
	HasToCreatedAt   bool
}

type AlertResolve struct {
	UserID  int64
	AlertID int64
}
